mne/epochs.py

```diff
@@ -128,7 +129,12 @@ def _save_split(epochs, fname, part_idx, n_parts):
 
     data *= decal[np.newaxis, :, np.newaxis]
 
-    write_float_matrix(fid, FIFF.FIFF_EPOCH, data)
+    if data.dtype == 'float64':
+        write_float_matrix(fid, FIFF.FIFF_EPOCH, data)
+    elif data.dtype == 'complex64':
+        write_complex_float_matrix(fid, FIFF.FIFF_EPOCH, data)
```

Comment:
Currently you should never hit this line because you have above `assert data.dtype in (`float64`, `complex128`).

To make this equivalent to `raw` you would want to do a new param `fmt='single' | 'double'` that defaults to `'single'` for backward compat (meaning float32/complex64 or float64/complex128 depending on real/complex data). It's probably worth adding here 1) to help avoid hitting file size limits when people know they don't need the extra precision, and 2) for consistency with `raw.save`.